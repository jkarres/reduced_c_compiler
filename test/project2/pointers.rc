int i;
int * pi;

function : void main() {
  pi = &i;

  *pi = 10;

  cout << *pi << endl;
  cout << i << endl;


  float f;
  float * pf = &f;

  *pf = *pi;

  cout << f << endl;
  cout << *pf << endl;
 

  int j = 42;

  int * pj = &j;

  int ** ppj = &pj;

  int *** pppj = &ppj;

  ++***pppj;

  cout << j << endl;
 
}
