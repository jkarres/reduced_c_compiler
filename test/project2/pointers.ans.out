10
10
10.00
10.00
43
